pkg/controllers/raycluster_controller.go

```diff
@@ -343,11 +344,11 @@ func desiredServiceAccount(cluster *rayv1.RayCluster) *corev1ac.ServiceAccountAp
 }
 
 func dashboardNameFromCluster(cluster *rayv1.RayCluster) string {
-	return "ray-dashboard-" + cluster.Name
+	return "ray-dashboard-" + cluster.Name + "-" + ControllerSpecificHash(cluster.Name, defaultHashLength)
 }
 
 func rayClientNameFromCluster(cluster *rayv1.RayCluster) string {
-	return "rayclient-" + cluster.Name
+	return "rayclient-" + cluster.Name + "-" + ControllerSpecificHash(cluster.Name, defaultHashLength)
```

Comment:
To go a step further, what about having a function is that file that returns the unique name, provided the resource name?

Reply:
so do something that basically does:
```
getUniqueName(name string) string {
  return name + "-" + ControllerSpecificHash(name)
}
```

Reply:
Yes, exactly.